go.sum

```diff
@@ -1314,53 +1389,77 @@ honnef.co/go/tools v0.0.1-2020.1.4 h1:UoveltGrhghAA7ePc+e+QYDHXrBps2PqFZiHkGR/xK
 honnef.co/go/tools v0.0.1-2020.1.4/go.mod h1:X/FiERA/W4tHapMX5mGpAtMSVEeEUOyHaw9vFzvIQ3k=
 honnef.co/go/tools v0.0.1-2020.1.5 h1:nI5egYTGJakVyOryqLs1cQO5dO0ksin5XXs2pspk75k=
 honnef.co/go/tools v0.0.1-2020.1.5/go.mod h1:X/FiERA/W4tHapMX5mGpAtMSVEeEUOyHaw9vFzvIQ3k=
-k8s.io/api v0.18.12 h1:97X6znOXMVgCKivTAgpBXGBGlCe3gbM++yFdldgBCaE=
-k8s.io/api v0.18.12/go.mod h1:3sS78jmUoGHwERyMbEhxP6owcQ77UxGo+Yy+dKNWrh0=
-k8s.io/apiextensions-apiserver v0.18.12 h1:b0jTgW/qwqZBMIJTMxkLvvAtNRDZboG5yZiIbOFgQv8=
-k8s.io/apiextensions-apiserver v0.18.12/go.mod h1:nihADkPed1L37Vxpz2/BrtxO9mCtINH23aNtUe/CRLo=
-k8s.io/apimachinery v0.18.12 h1:bLFXU4IxOu06F6Z6PV7eqtapXFb1G2q0ni0XBNFtJH8=
-k8s.io/apimachinery v0.18.12/go.mod h1:PF5taHbXgTEJLU+xMypMmYTXTWPJ5LaW8bfsisxnEXk=
+k8s.io/api v0.18.8/go.mod h1:d/CXqwWv+Z2XEG1LgceeDmHQwpUJhROPx16SlxJgERY=
+k8s.io/api v0.19.7 h1:MpHhls03C2pyzoYcpbe4QqYiiZjdvW+tuWq6TbjV14Y=
+k8s.io/api v0.19.7/go.mod h1:KTryDUT3l6Mtv7K2J2486PNL9DBns3wOYTkGR+iz63Y=
+k8s.io/apiextensions-apiserver v0.19.7 h1:aV9DANMSCCYBEMbtoT/5oesrtcciQrjy9yqWVtZZL5A=
+k8s.io/apiextensions-apiserver v0.19.7/go.mod h1:XJNNtjISNNePDEUClHt/igzMpQcmjVVh88QH+PKztPU=
+k8s.io/apimachinery v0.0.0-20190703205208-4cfb76a8bf76/go.mod h1:M2fZgZL9DbLfeJaPBCDqSqNsdsmLN+V29knYJnIXlMA=
+k8s.io/apimachinery v0.18.8/go.mod h1:6sQd+iHEqmOtALqOFjSWp2KZ9F0wlU/nWm0ZgsYWMig=
```

Comment:
Maybe we should not get rid of the replace in `go.mod` because it seems like we still import 0.18.8 like this.

Reply:
the `go.sum` file is just a list of modules and hash we've seen

what's being used is defined in `go.mod` you can see what's being used in the `vendor/modules.txt` file

Reply:
This is not entirely accurate AFAIK - after running go mod tidy the `go.sum` contains direct dependencies but also transitive dependencies. In this case `v0.18.8` is most likely a transitive dependency (we can probably find out why with go mod graph).

You are correct however, it should be using the newer version. These transitive dependencies are only a problem if we run security scanners such as nancy (see here for example: https://github.com/sonatype-nexus-community/nancy/issues/152#issuecomment-675716615 ).

TL;DR you are correct it should not matter as we do not use such scanners as far as I can find.

Reply:
oh interesting - I think it's https://github.com/google/go-containerregistry/blob/main/pkg/authn/k8schain/go.mod#L6

Reply:
let me fix that

Reply:
oddly I don't see v0.18.x in the vendored folder modules.txt - which I believe takes precedent over the module cache

Reply:
I'm sorta confused now 

```
$ go list -m all  | grep v0.18
github.com/aws/aws-sdk-go-v2 v0.18.0
github.com/lightstep/lightstep-tracer-go v0.18.1
k8s.io/cloud-provider v0.18.8
k8s.io/csi-translation-lib v0.18.8
k8s.io/legacy-cloud-providers v0.18.8
```

Is says there's a dep

but when you go to each binary in this repo it says it's not used 

```
dmad:controller dprotasowski$ go mod why -m k8s.io/cloud-provider
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/controller/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/eventlistenersink/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/webhook/kodata/third_party
# k8s.io/cloud-provider
(main module does not need module k8s.io/cloud-provider)
dmad:controller dprotasowski$ go mod why -m k8s.io/cloud-provider
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/controller/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/eventlistenersink/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/webhook/kodata/third_party
# k8s.io/cloud-provider
(main module does not need module k8s.io/cloud-provider)
dmad:controller dprotasowski$ cd ../binding-eval/
dmad:binding-eval dprotasowski$ go mod why -m k8s.io/cloud-provider
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/controller/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/eventlistenersink/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/webhook/kodata/third_party
# k8s.io/cloud-provider
(main module does not need module k8s.io/cloud-provider)
dmad:binding-eval dprotasowski$ cd ../triggerrun/
dmad:triggerrun dprotasowski$ go mod why -m k8s.io/cloud-provider
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/controller/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/eventlistenersink/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/webhook/kodata/third_party
# k8s.io/cloud-provider
(main module does not need module k8s.io/cloud-provider)
dmad:triggerrun dprotasowski$ cd ../interceptors/
dmad:interceptors dprotasowski$ go mod why -m k8s.io/cloud-provider
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/controller/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/eventlistenersink/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/webhook/kodata/third_party
# k8s.io/cloud-provider
(main module does not need module k8s.io/cloud-provider)
dmad:interceptors dprotasowski$ cd ../eventlistenersink/
dmad:eventlistenersink dprotasowski$ go mod why -m k8s.io/cloud-provider
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/controller/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/eventlistenersink/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/webhook/kodata/third_party
# k8s.io/cloud-provider
(main module does not need module k8s.io/cloud-provider)
dmad:eventlistenersink dprotasowski$ cd ../webhook/
dmad:webhook dprotasowski$ go mod why -m k8s.io/cloud-provider
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/controller/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/eventlistenersink/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/webhook/kodata/third_party
# k8s.io/cloud-provider
(main module does not need module k8s.io/cloud-provider)
dmad:webhook dprotasowski$ export GO111MODULE=on
dmad:webhook dprotasowski$ go mod why -m k8s.io/cloud-provider
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/controller/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/eventlistenersink/kodata/third_party
warning: ignoring symlink /Users/dprotasowski/go/src/github.com/tektoncd/triggers/cmd/webhook/kodata/third_party
# k8s.io/cloud-provider
(main module does not need module k8s.io/cloud-provider)
```

Reply:
I see where it's coming from

ie the graph
```
github.com/google/go-containerregistry/pkg/authn/k8schain@v0.0.0-20210127165842-51f01e739161 github.com/google/go-containerregistry@v0.3.0
github.com/google/go-containerregistry/pkg/authn/k8schain@v0.0.0-20210127165842-51f01e739161 github.com/vdemeester/k8s-pkg-credentialprovider@v1.18.1-0.20201019120933-f1d16962a4db
```

Reply:
I'm guessing the modules have dependencies but the specific packages aren't being used so it doesn't factor into the version selected?

Reply:
Sorry for creating this rabbit hole for you - yes 0.18.X is not actually used in the compiled code. My initial question was just if we in tekton / this project care if a transitive dependency import was still pointing elsewhere. Actual compiled code will all use 0.20.X

This section also explains how always the higher version is taken in these cases: https://golang.org/ref/mod#major-version-suffixes

So yeah, sorry again for the confusion this caused :)

Reply:
No worries - it forces us to clean up a bunch of things upstream